### PR TITLE
Fixed Noclip and Clip command

### PR DESCRIPTION
**Pull request by kaiserandaxl**
I made both the Noclip and Clip command be more readable and actually work.

### CODE REVIEW

**Comment by karl-police** [comment]
I tested your changes, and ``:unnoclip`` still let me walk through walls.

File: MainModule/Server/Dependencies/Assets/Clip.client.lua [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+for _,v in pairs(script.Parent:GetDescendants()) do
+	if v:IsA("BasePart") and (v.Name == "HumanoidRootPart" or v.Name == "UpperTorso" or v.Name == "LowerTorso" or v.Name == "Torso" or (v.Name == "Head" and script.Parent:FindFirstChild("Torso"))) then
+		v.CanCollide = true
+	end
+end
+
+Humanoid:ChangeState(18)		--| Enum.HumanoidStateType.None (Used so we can remove the Enum.HumanoidStateType.StrafingNoPhysics)
+task.wait()
+script:Destroy()
```
Comment (karl-police) [inline MainModule/Server/Dependencies/Assets/Clip.client.lua:11]: Is this even being used?

**Review by karl-police — CHANGES_REQUESTED** [review]
What is "Clip" for?
Is it just to unnoclip?

ADONIS_NoClip is not inside Humanoid, so it's always ``nil``, and that's why unclip doesn't work.

You should change this, I just got ``:unnoclip`` to work by changing that.

I don't see what ``:clip`` is supposed to even do to see differences.

**Comment by kaiserandaxl** [comment]
The reason it won't work is because the Clip script isn't yet in the dependencies, I have made those changes to where if Clip was inside of dependencies

@karl-police

**Comment by kaiserandaxl** [comment]
Also to answer your question about how this fixes it is this.

Before it was deleting the the script that changed the players state and made them have the ability to walk through walls, well this doesn't work anymore and they still are able to walk through walls so I made one extra script

This other script is supposed to give them the ability to collide with walls and objects again and then change the HumanoidState to 18 (none) which then gives roblox the abiltiy to set the state to whatever they want.

**Comment by karl-police** [comment]
> The reason it won't work is because the Clip script isn't yet in the dependencies, I have made those changes to where if Clip was inside of dependencies
> 
> @karl-police

I put it in there on Studio when testing for myself.

**Comment by karl-police** [comment]
Unnoclip still won't work because the code is still looking inside Humanoid.

**Comment by kaiserandaxl** [comment]
Fixed it

**Comment by karl-police** [comment]
> Fixed it

Did you test it as well?

**Review by karl-police — CHANGES_REQUESTED** [review]
prolly you wanna fix it

**Comment by kaiserandaxl** [comment]
@karl-police you gonna look and find any errors

**Review by karl-police — APPROVED** [review]

**Review by karl-police — CHANGES_REQUESTED** [review]
Instead of renaming "Clip" I think you renamed "Clipper" to "DisableNoclip", but there's still code that need that dependency "Clipper".

Rename DisableNoclip back to Clipper. And re-name "Clip" to DisableNoclip instead.

**Review by karl-police — APPROVED** [review]

**Comment by karl-police** [comment]
@Sceleratis 
@Expertcoderz 

It fixes Noclip and Clip aka. Unnoclip

**Review by WalkerOfBacon — COMMENTED** [review]
This works however the notification doesnt show up when running both :noclip and :clip

**Comment by kaiserandaxl** [comment]
Do you have command feedback enabled in settings

@WalkerOfBacon

**Comment by karl-police** [comment]
@Expertcoderz I tested it, you can merge it.

**Comment by WalkerOfBacon** [comment]
> Do you have command feedback enabled in settings
> 
> @WalkerOfBacon

Not sure

**Comment by LiveFireExercise** [comment]
Gonna test this one sec

**Comment by LiveFireExercise** [comment]
image
yeah uh

**Comment by LiveFireExercise** [comment]
Also this PR has some weird effects on the :fly command, might wanna look at that

**Comment by watameln** [comment]
why not just use the kohls admin noclip and clip :trollface:

**Review by squiddickod — APPROVED** [review]

**Comment by Dimenpsyonal** [comment]
Idk why you approved the changes because it doesn't work

**Review by WalkerOfBacon — APPROVED** [review]
@Dimenpsyonal uhhhm actually it works

**Comment by WalkerOfBacon** [comment]
and fly works normally too

**Comment by WalkerOfBacon** [comment]
fix lint smh

**Comment by kaiserandaxl** [comment]
Is it gonna be put in yes or no, if not tell me the error so it can be fixed and the error removed

**Comment by Dimenpsyonal** [comment]
If it works then I'll put it in :)

**Comment by ccuser44** [comment]
image
The lint errors do not come as a part of this pull

**Comment by Dimenpsyonal** [comment]
Yes

**Comment by Dimenpsyonal** [comment]
Make sure to include proof of functionality :)

**Review by SuperCater — APPROVED** [review]

**Pull request closed without merging**